fishnet.py

```diff
@@ -1714,6 +1789,101 @@ def cmd_run(args):
         print("Ignored deprecated option --threads. Did you mean --cores?")
         print()
 
+    return cores, threads, instances, memory, user_backlog, system_backlog
+
+
+def cmd_benchmark(args):
+    conf = load_conf(args)
+    cores, threads, instances, memory, _, _ = display_config(args, conf)
+
+    buckets = [0] * instances
+    for i in range(0, cores):
+        buckets[i % instances] += 1
+
+    instance_memory = memory // instances
+    workers = [BenchmarkWorker(conf, bucket, instance_memory) for bucket in buckets]
+
+    logging.info("Starting %d workers" % len(workers))
+
+    # Start all threads
+    for i, worker in enumerate(workers):
+        worker.set_name("><> %d" % (i + 1))
+        worker.setDaemon(True)
+        worker.start()
+
+    # Note: Waiting with timeout on worker.running would be nicer.
+    time.sleep(3)
+    for worker in workers:
+        if not worker.running:
+            logging.warning("Worker %s never started", worker.name)
+    logging.info("All workers started")
+
+    # Let SIGTERM and SIGINT gracefully terminate the program
+    handler = SignalHandler()
```

Comment:
Nit: `SignalHandler()` should only be created in a `try` block that can catch its signals and always finally be defused (`handler.ignore = True`)

Reply:
I moved the `SignalHandler()` into the try block.
I also added the `handler.ignore`, I will have another PR after this that I believe cleans up handler in `cmd_run`